twenty-first/src/shared_math/polynomial.rs

```diff
@@ -336,177 +206,43 @@ impl<PFElem: FiniteField> Polynomial<PFElem> {
             coefficients: squared_coefficients,
         }
     }
-}
 
-impl<PFElem: FiniteField> Polynomial<PFElem> {
-    pub fn are_colinear(points: &[(PFElem, PFElem)]) -> bool {
-        if points.len() < 3 {
-            println!("Too few points received. Got: {} points", points.len());
-            return false;
+    #[must_use]
+    pub fn fast_mod_pow(&self, pow: BigInt) -> Self {
```

Comment:
Why `pow: BigInt` as opposed to `PFElem`, `u64`, `u32` or `u8`?

Reply:
This is just code I moved around, since the MulAssign<BFieldElement> trait requirement had to be added to some methods.